verticapy/pipeline/parser.py

```diff
@@ -0,0 +1,207 @@
+affiliates.  Licensed  under  the   Apache  License,
+Version 2.0 (the  "License"); You  may  not use this
+file except in compliance with the License.
+
+You may obtain a copy of the License at:
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless  required  by applicable  law or  agreed to in
+    writing, software  distributed  under the  License is
+    distributed on an  "AS IS" BASIS,  WITHOUT WARRANTIES
+    OR CONDITIONS OF ANY KIND, either express or implied.
+    See the  License for the specific  language governing
+    permissions and limitations under the License.
+"""
+"""
+This script runs the Vertica Machine Learning Pipeline Parser.
+"""
+
+# base package imports
+import yaml
+import argparse
+from tqdm import tqdm
+import re
+
+# required packages
+import verticapy as vp
+from verticapy.datasets import (
+    load_airline_passengers,
+    load_amazon,
+    load_cities,
+    load_commodities,
+    load_gapminder,
+    load_iris,
+    load_laliga,
+    load_market,
+    load_pop_growth,
+    load_smart_meters,
+    load_titanic,
+    load_winequality,
+    load_world,
+    load_africa_education,
+)
+from verticapy._utils._sql._sys import _executeSQL
+
+# local
+from ._helper import required_keywords, execute_and_return, setup
+from . import _ingest
+from . import _transform
+from . import _train
+from . import _validate
+from . import _schedule
+
+SUPPORTED_DATASETS = [
+    load_airline_passengers,
+    load_amazon,
+    load_cities,
+    load_commodities,
+    load_gapminder,
+    load_iris,
+    load_laliga,
+    load_market,
+    load_pop_growth,
+    load_smart_meters,
+    load_titanic,
+    load_winequality,
+    load_world,
+    load_africa_education,
+]
+
+parser = argparse.ArgumentParser(
+    description="""Vertica Pipelines is an open source platform for
+    managing data scientists machine learning pipelines.
+    They are built on a human-readable data format: YAML."""
+)
+parser.add_argument("connection_file", help="Path to the connection yaml file")
+parser.add_argument("input_file", help="Path to the input yaml file")
+parser.add_argument(
+    "-o",
+    "--outfile",
+    nargs="?",
+    help="[Optional] Path to the output sql file. If unspecified pipeline_name.sql will be used.",
+    default=None,
+    required=False,
+)
+
+args = parser.parse_args()
+config_name = args.connection_file
+file_name = args.input_file
+output_name = args.outfile
+
+with open(config_name, "r", encoding="utf-8") as file:
+    connect = yaml.safe_load(file)
+    if required_keywords(connect, ["host", "port", "database", "password", "user"]):
+        vp.new_connection(
+            {
+                "host": connect["host"],
+                "port": connect["port"],
+                "database": connect["database"],
+                "password": connect["password"],
+                "user": connect["user"],
+            },
+            name="temp",
+        )
+        vp.connect("temp")
+
```

Comment:
Shouldn't we raise an error is some connection parameters are missing?

Reply:
An error is raised:
  File "/VerticaPy/verticapy/pipeline/parser.py", line 96, in <module>
    if required_keywords(connect, ["host", "port", "database", "password", "user"]):
  File "/VerticaPy/verticapy/pipeline/_helper.py", line 31, in required_keywords
    raise KeyError(f"Missing keyword(s): '{keyword}'")
KeyError: "Missing keyword(s): 'password'"